### PR TITLE
feat: use napi-rs blst bindings

### LINKED ISSUE #6462: Add randomizing factors to bls same message verification
### Problem description

Right now when we validate bls signatures of the same message, we aggregate signatures and public keys, if the aggregated signature is verified we consider all signatures are verified. But there's a chance 2 signatures are invalid and final verification result is still `true`.

### Solution description

Add randomizing factors following this approach https://ethresear.ch/t/fast-verification-of-multiple-bls-signatures/5407

quick benchmark:

```typescript
itBench({
      id: `BLS same message - randomizing factor - ${count} - ${bls.implementation}`,
      beforeEach: () => linspace(0, count - 1).map((i) => getSetSameMessage(i)),
      fn: (sets) => {
        const randomizingFactors = [];
        for (let i = 0; i < sets.length; i++) {
          randomizingFactors.push(randomBytesNonZero(8));
        }
        const PkConstructor = blst.P1;
        const aggregatedPubkey = new PkConstructor(); // Faster than using .dup()
        for (let i = 0; i < sets.length; i++) {
          const clone = (sets[i].publicKey as unknown as BlstPublicKey).value.dup();
          const pk = new BlstPublicKey(clone).jacobian.mult(randomizingFactors[i]);
          aggregatedPubkey.add(pk);
        }

        const SigConstructor = blst.P2;
        const aggregatedSignature = new SigConstructor(); // Faster than using .dup()
        for (let i = 0; i < sets.length; i++) {
          const clone = (bls.Signature.fromBytes(sets[i].signature, CoordType.affine, true) as unknown as BlstSignature).value.dup();
          const sig = new BlstSignature(clone).jacobian.mult(randomizingFactors[i]);
          aggregatedSignature.add(sig);
        }
        const isValid = verify(sets[0].message, new BlstPublicKey(aggregatedPubkey), new BlstSignature(aggregatedSignature));
        if (!isValid) throw Error("Invalid");
      },
    });
```

shows this is 2x slower than the current approach but it's still ~2.65x faster than `verifyMultipleSignatures`

### Additional context

Given signature aggregation already took 25% of a mainnet node, it's not sustainable to still do this in main thread, we should offload the work to worker/libuv instead

Also right now we queue attestation messages in `IndexedGossipQueue` and bls queue, we should only queue in `IndexedGossipQueue`. For bls, we need to write new worker api (or libuv implementation) and verify same message signatures immediately without queueing.

cc @matthewkeil

### PR DESCRIPTION
**Pull request by wemeetagain**
**Motivation**

- we want to eventually use a newer blst binding
- see https://github.com/ChainSafe/lodestar/pull/6853

**Description**

- use https://github.com/ChainSafe/blst-ts/pull/149
- deployed to feat2

Closes #6462

### CODE REVIEW

**Comment by wemeetagain** [comment]
ready for re-review, rebased off unstable and deployed to feat2

File: packages/beacon-node/src/chain/bls/multithread/jobItem.ts [hunk 1]
```diff
@@ -70,15 +69,15 @@ export function jobItemWorkReq(job: JobQueueItem, format: PointFormat, metrics:
       // and not a problem in the near future
       // this is monitored on v1.11.0 https://github.com/ChainSafe/lodestar/pull/5912#issuecomment-1700320307
```
Comment (twoeths) [inline packages/beacon-node/src/chain/bls/multithread/jobItem.ts]: the comment is not relevant anymore, need to update it with test result on feat2

File: packages/beacon-node/src/chain/bls/multithread/jobItem.ts [hunk 2]
```diff
@@ -70,15 +69,15 @@ export function jobItemWorkReq(job: JobQueueItem, format: PointFormat, metrics:
...
       const timer = metrics?.blsThreadPool.signatureDeserializationMainThreadDuration.startTimer();
```
Comment (twoeths) [inline packages/beacon-node/src/chain/bls/multithread/jobItem.ts]: this metric is misleading now, need to change it to a new name and description, and update grafana bls dashboard

File: packages/beacon-node/src/chain/bls/multithread/jobItem.ts (hunk 1 above)
Comment (twoeths) [reply packages/beacon-node/src/chain/bls/multithread/jobItem.ts]: `aggregateWithRandomness` now takes ~2s vs ~1.78s on stable mainnet node, just something to note

Screenshot 2024-07-23 at 10 01 47

**Comment by twoeths** [comment]
metrics look comparable to stable mainnet node in terms of:
- gossipsub: mesh peers, gossip traffic, forward peers
- attestation job time + job wait time

and that's with randomness added so it's an awesome result. Some other metrics I found it different to stable:
- signature set verification time / set is 1.5x faster
- from worker latency is 1.5x slower

CPU pause time rate is awesome (scavenge is ~2.3 % compared to 2.7% on stable and unstable)

Screenshot 2024-07-23 at 10 11 40

same to lg1k, it's ~0.75% vs ~0.89% which lead to greater epoch transition time (~2.6s on stable 1k and ~2 to ~2.3s on unstable 1k)

Screenshot 2024-07-23 at 10 15 17

File: packages/beacon-node/src/chain/bls/multithread/jobItem.ts [hunk 3]
```diff
@@ -70,15 +69,15 @@ export function jobItemWorkReq(job: JobQueueItem, format: PointFormat, metrics:
...
-      const signatures = job.sets.map((set) => bls.Signature.fromBytes(set.signature, CoordType.affine, true));
+      const {pk, sig} = aggregateWithRandomness(job.sets.map((set) => ({pk: set.publicKey, sig: set.signature})));
```
Comment (twoeths) [inline packages/beacon-node/src/chain/bls/multithread/jobItem.ts:73]: `gc` improved a lot 🚀 , I guess it's thanks to this `napi-rs` binding? @wemeetagain @matthewkeil

File: dashboards/lodestar_block_processor.json [hunk 4]
```diff
@@ -5101,7 +5101,7 @@
             "uid": "${DS_PROMETHEUS}"
           },
           "editorMode": "code",
-          "expr": "rate(lodestar_bls_thread_pool_signature_deserialization_main_thread_time_seconds_bucket[$rate_interval])",
```
Comment (twoeths) [inline dashboards/lodestar_block_processor.json:5104]: this one and the below is not right, should be replaced with `lodestar_bls_thread_pool_aggregate_with_randomness_main_thread_time_seconds`

Comment (twoeths) [reply dashboards/lodestar_block_processor.json:5104]: nvm the old code was not right

**Review by twoeths — APPROVED** [review]
persist "Attnet count with > 0 mesh peers" from the mainnet node deploying this branch

Screenshot 2024-07-25 at 14 12 52

vs unstable/stable mainnet node
Screenshot 2024-07-25 at 14 13 09

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.21.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.21.0) :tada:

**Pull request merged**